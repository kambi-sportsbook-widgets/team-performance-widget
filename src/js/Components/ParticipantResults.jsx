import React from 'react';
import PropTypes from 'prop-types';

const ParticipantResults = ({ children, name }) => {
   return (
      <div className='kw-participantresults'>
         <div className='kw-participantresults__header KambiWidget-secondary-header KambiWidget-card-inner-border'>{name}</div>
         {children}
      </div>
   );
};

ParticipantResults.propTypes = {
   /**
    * Inner components
    */
   children: PropTypes.node.isRequired,

   /**
    * Participant name
    */
   name: PropTypes.string.isRequired
};

export default ParticipantResults;
