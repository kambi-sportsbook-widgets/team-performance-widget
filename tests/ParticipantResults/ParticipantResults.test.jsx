/* eslint-env jest */
import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import ParticipantResults from '../../src/js/Components/ParticipantResults';

let renderer;

describe('ParticipantResults DOM rendering', () => {
   beforeEach(() => {
      renderer = ReactTestUtils.createRenderer();
   });
   it('renders correctly', () => {
      expect(renderer.render(
         <ParticipantResults name='TheName'>
            <span>content</span>
         </ParticipantResults>
      )).toMatchSnapshot();
   });
});
