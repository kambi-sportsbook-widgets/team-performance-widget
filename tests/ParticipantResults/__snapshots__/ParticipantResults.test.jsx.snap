// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantResults DOM rendering renders correctly 1`] = `
<div
  className="kw-participantresults"
>
  <div
    className="kw-participantresults__header KambiWidget-secondary-header KambiWidget-card-inner-border"
  >
    TheName
  </div>
  <span>
    content
  </span>
</div>
`;
